hal/include/HAL/Digital/PWM.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <stdint.h>
+
+extern "C"
+{
+  bool checkPWMChannel(void* digital_port_pointer);
```

Comment:
I don't have a great answer on why they are all void*s. I did get the approval to make these use typed pointers instead of void pointers, but that change will have to wait until the resource change, as its going to break alot and I would rather change all those things at once.